crates/solver-service/src/apis/quote/custody.rs

```diff
@@ -62,11 +63,19 @@ pub enum CustodyDecision {
 }
 
 /// Custody strategy decision engine
-pub struct CustodyStrategy {}
+pub struct CustodyStrategy {
+	/// Cached token capabilities to avoid repeated lookups
+	token_capabilities: HashMap<String, TokenCapabilities>,
+	/// Permit2 contract addresses per chain
+	permit2_addresses: HashMap<u64, Address>,
```

Comment:
These are currently not used. Lookups are in/memory.. not any network or RPC calls. So caching here doesn't make sense iiuc.